clippy_lints/src/methods/unnecessary_lazy_eval.rs

```diff
@@ -20,8 +20,9 @@ pub(super) fn check<'tcx>(
 ) {
     let is_option = is_type_diagnostic_item(cx, cx.typeck_results().expr_ty(recv), sym::Option);
     let is_result = is_type_diagnostic_item(cx, cx.typeck_results().expr_ty(recv), sym::Result);
+    let is_bool = is_type_diagnostic_item(cx, cx.typeck_results().expr_ty(recv), sym::bool);
```

Comment:
```suggestion
    let is_bool = cx.typeck_results().expr_ty(recv).is_bool();
```

There's no `bool` diagnostic item :slightly_smiling_face: